### PR TITLE
Remove panic in storage and add log

### LINKED ISSUE #79835: Remove panic in staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go
Remove panic in [staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go#L188), please return error instead following [golang CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments#dont-panic).

### PR DESCRIPTION
**Pull request by caohe**
Signed-off-by: caohe <caohe9603@gmail.com>

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Remove panic in staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go.
Add some log if versioner.UpdateObject() returns error.

**Which issue(s) this PR fixes**:

Fixes #79835

**Special notes for your reviewer**:

/sig api-machinery

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

**Comment by caohe** [comment]
@answer1991 PTAL

**Comment by caohe** [comment]
@yue9944882 PTAL

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go [hunk 1]
```diff
@@ -742,7 +742,10 @@ func (s *store) getStateFromObject(obj runtime.Object) (*objState, error) {
 	if err != nil {
 		return nil, err
 	}
-	s.versioner.UpdateObject(state.obj, uint64(rv))
+	err = s.versioner.UpdateObject(state.obj, uint64(rv))
+	if err != nil {
+		klog.Errorf("failed to update object: %v", err)
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go]: failed to update object version

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go [hunk 2]
```diff
@@ -742,7 +742,10 @@ func (s *store) getStateFromObject(obj runtime.Object) (*objState, error) {
 	if err != nil {
 		return nil, err
 	}
-	s.versioner.UpdateObject(state.obj, uint64(rv))
+	err = s.versioner.UpdateObject(state.obj, uint64(rv))
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go]: if err := s.versioner...; err != nil {

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go [hunk 3]
```diff
@@ -779,14 +782,17 @@ func (s *store) ttlOpts(ctx context.Context, ttl int64) ([]clientv3.OpOption, er
...
-		panic("unable to convert output object to pointer")
+		return fmt.Errorf("unable to convert output object to pointer: %v", err)
 	}
 	_, _, err := codec.Decode(value, nil, objPtr)
 	if err != nil {
 		return err
 	}
 	// being unable to set the version does not prevent the object from being extracted
-	versioner.UpdateObject(objPtr, uint64(rev))
+	err = versioner.UpdateObject(objPtr, uint64(rev))
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go]: if err := versioner...; err != nil {

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go [hunk 4]
```diff
@@ -779,14 +782,17 @@ func (s *store) ttlOpts(ctx context.Context, ttl int64) ([]clientv3.OpOption, er
...
+	if err != nil {
+		klog.Errorf("failed to update object: %v", err)
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go]: failed to update object version

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go [hunk 5]
```diff
@@ -797,7 +803,10 @@ func appendListItem(v reflect.Value, data []byte, rev uint64, pred storage.Selec
 		return err
 	}
 	// being unable to set the version does not prevent the object from being extracted
-	versioner.UpdateObject(obj, rev)
+	err = versioner.UpdateObject(obj, rev)
+	if err != nil {
+		klog.Errorf("failed to update object: %v", err)
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go]: failed to update object version

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go [hunk 6]
```diff
@@ -797,7 +803,10 @@ func appendListItem(v reflect.Value, data []byte, rev uint64, pred storage.Selec
 		return err
 	}
 	// being unable to set the version does not prevent the object from being extracted
-	versioner.UpdateObject(obj, rev)
+	err = versioner.UpdateObject(obj, rev)
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go]: if err := versioner...; err != nil {

**Review by wojtek-t — COMMENTED** [review]
couple nits

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go (hunk 1 above)
Comment (caohe) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go]: done. thx.

Comment (answer1991) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go]:
```go
if err := s.versioner.UpdateObject(state.obj, uint64(rv)); err != nil {
...
}
```

Comment (caohe) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go]: @answer1991 thx. I have already moved line 745 into the `if` clause. I suppose my code above may be outdated.

**Comment by wojtek-t** [comment]
@caohe - please squash the commits and I will approve.

**Comment by caohe** [comment]
@wojtek-t done. PTAL

**Pull request merged**